server/odc-service/src/main/java/com/oceanbase/odc/config/HookConfiguration.java

```diff
@@ -142,18 +144,22 @@ private void approvalFlowConfigUsageCheck(Long flowConfigId) {
     }
 
     private void projectReferenceCheck(Long userId, Long organizationId) {
+        Map<Long, ProjectEntity> id2Project =
+                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> !p.getArchived())
+                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
+
         Map<Long, Set<ResourceRoleName>> projectId2ResourceRoleNames =
                 resourceRoleService.getProjectId2ResourceRoleNames(userId).entrySet().stream()
                         .filter(e -> e.getValue().contains(ResourceRoleName.OWNER)
                                 || e.getValue().contains(ResourceRoleName.DBA))
+                        .filter(e -> id2Project.containsKey(e.getKey()))
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
         if (projectId2ResourceRoleNames.size() == 0) {
             return;
         }
-        Map<Long, ProjectEntity> id2Project =
-                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> p.getArchived() == false)
-                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
-        String names = projectId2ResourceRoleNames.keySet().stream().map(id -> id2Project.get(id).getName())
+        String names = projectId2ResourceRoleNames.keySet().stream()
+                .map(id -> Optional.ofNullable(id2Project.get(id))
+                        .map(ProjectEntity::getName).orElseThrow(() -> new UnexpectedException("NullPointerException")))
```

Comment:
The exception error message is supposed to be readable, but `NullPointerException` does not bring any useful information.
Message suggestion: "project not found, project id=xxx". Or as I said in the previous comment, you could always assume you would never get `null` as you have already filtered out these projects, you may just delete these new codes.

Reply:
got it